src/galaxy/StarSystemGenerator.cpp

```diff
@@ -1259,6 +1259,122 @@ void StarSystemRandomGenerator::MakeBinaryPair(SystemBody *a, SystemBody *b, fix
 	b->m_orbMax = orbMax;
 }
 
+SystemBody* StarSystemRandomGenerator::MakeSystemBody(RefCountedPtr<StarSystem::GeneratorAPI> system, const std::string &systemName, const std::string &bodySuffix)
+{
+	SystemBody *newBody = system->NewBody();
+	newBody->m_name = systemName;
+	if (!bodySuffix.empty()) {
+		newBody->m_name += " ";
+		newBody->m_name += bodySuffix;
+	}
+
+	return newBody;
+}
+
+SystemBody* StarSystemRandomGenerator::MakeGravPointForBodies(RefCountedPtr<StarSystem::GeneratorAPI> system, const std::string &systemName, SystemBody * body1, SystemBody * body2)
+{
+	const std::string suffix1 = body1->m_name.substr(systemName.size() + 1);
+	const std::string suffix2 = body2->m_name.substr(systemName.size() + 1);
+	const std::string suffixGrav = suffix1 + ',' + suffix2;
+
+	SystemBody *grav = MakeSystemBody(system, systemName, suffixGrav);
+	grav->m_type = SystemBody::TYPE_GRAVPOINT;
+	grav->m_mass = body1->GetMassAsFixed() + body2->GetMassAsFixed();
+
+	body1->m_parent = grav;
+	body2->m_parent = grav;
+
+	grav->m_children.push_back(body1);
+	grav->m_children.push_back(body2);
+
+	return grav;
+}
+
+char StarSystemRandomGenerator::GetStarSuffixFromOffset(const int offsetId) const
+{
+	return 'A' + Clamp(offsetId, 0, 26);
+}
+
+SystemBody* StarSystemRandomGenerator::PlaceStars(Random &rng, RefCountedPtr<StarSystem::GeneratorAPI> system, const Sector::System &secSys, const int offset, const int numStars, const fixed maxMass = fixed())
+{
+	const std::string systemName = secSys.GetName();
+	SystemBody *body;
+
+	if (numStars == 1) {
+		// We have only one star in the system, make it root
+
+		SystemBody::BodyType type = secSys.GetStarType(0);
+		const std::string suffix(1, GetStarSuffixFromOffset(offset));
+		body = MakeSystemBody(system, systemName, suffix);
```

Comment:
Something I did just spot, this shouldnt be taking any suffix at all. If there's a single star in a system then it's just the systemName

Reply:
> Something I did just spot, this shouldnt be taking any suffix at all. If there's a single star in a system then it's just the systemName

It is called recursively, empty root suffix is handled in `StarSystemRandomGenerator::Apply` after calling PlaceStars

Reply:
so name it, then always override the name later?

Reply:
> so name it, then always override the name later?

yes. It was easier than pre-naming checks.